tests/e2e/asserts_test.go

```diff
@@ -2234,14 +2236,23 @@ func collectAndAssertDefaultMetricsPresentOnEachPod(namespace, clusterName, curl
 	})
 }
 
+// CreateResourceFromFile creates the Kubernetes objects defined in a YAML sample file
 func CreateResourceFromFile(namespace, sampleFilePath string) {
-	Eventually(func() error {
-		_, _, err := testsUtils.RunUnchecked("kubectl apply -n " + namespace + " -f " + sampleFilePath)
-		if err != nil {
-			return err
+	objects, err := testsUtils.CreateObjectsFromYAML(sampleFilePath, namespace)
+	Expect(err).ToNot(HaveOccurred())
+	for _, obj := range objects {
+		if cluster, ok := obj.(*apiv1.Cluster); ok {
+			storageClassName := os.Getenv("E2E_DEFAULT_STORAGE_CLASS")
```

Comment:
Perhaps we could replace the environment variables such as `E2E_DEFAULT_STORAGE_CLASS` before parsing the YAML. Doing that every resource would be handled in the same way.